app/services/katello/pulp3/api/core.rb

```diff
@@ -47,6 +47,10 @@ def remotes_uln_api
           fail NotImplementedError
         end
 
+        def get_remotes_api(*)
```

Comment:
I am thinking how to communicate that we expect a hash-parameter containing either `:href` or `:url`, here.
Would a comment be sufficient or do we want to specify the parameter list as required.
The latter will probably make rubocop complain about it.

Reply:
I would argue, that in this parent class, we don't expect any particular parameters here.

The yum type version of this method now start with:
```
        def get_remotes_api(opts = {})
          fail 'Provide exactly one of href or url for yum remote selection!' unless !opts[:href] != !opts[:url]
```
This code makes pretty clear what type of arguments we expect in the yum case.

So maybe we can leave things as is? In case developers working on other content types wonder why we need the core version of this method at all, they can always use `git grep 'def get_remotes_api'` or `git grep get_remotes_api` to study up.

Reply:
> I would argue, that in this parent class, we don't expect any particular parameters here.

The problem is other content-types that might need to implement their own `get_remotes_api()` are not free to choose different parameters, because the method is called with `:href` or `:url` in the generic code (like app/services/katello/pulp3/repository.rb).

Therefore, the parent-method must IMHO define what can be evaluated. It is always possible to look for this in existing code that uses it or defines child-methods, but the first location I would look for is the method-definition in the parent-class.

Reply:
If another content type might want to add a second remote type some time in the future, there is no guarantee that that content types discriminator for choosing the remote type would also be the URL. It follows that there would likely need to be some refactoring/extending at that point. This coupled with the fact that `*` seems semantically correct to me (for the parent class), I feel like the best way to address your concern might really be a comment. Something along the lines of:
```
# This function is currently called with either opts[:url] or opts[:href] parameters for the benefit of the yum content type.
```
Or maybe simply:
```
# See the yum version of this function to see how it can be used.
```
What do you think?

Reply:
Added a comment.